openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -235,33 +230,28 @@ def update_email_opt_in(user, org, optin):
     Returns:
         None
 
+    Raises:
+         UserNotFound: no user profile exists for the specified user.
     """
-    # Avoid calling get_account_settings because it introduces circularity for many callers who need both
-    # preferences and account information.
     try:
         user_profile = UserProfile.objects.get(user=user)
     except ObjectDoesNotExist:
         raise UserNotFound()
-
-    year_of_birth = user_profile.year_of_birth
-    of_age = (
-        year_of_birth is None or  # If year of birth is not set, we assume user is of age.
-        datetime.datetime.now(UTC).year - year_of_birth >  # pylint: disable=maybe-no-member
-        getattr(settings, 'EMAIL_OPTIN_MINIMUM_AGE', 13)
+    preference, _ = UserOrgTag.objects.get_or_create(
+        user=user, org=org, key='email-optin'
     )
-
+    user_requires_parental_consent = user_profile.requires_parental_consent(
+        age_limit=getattr(settings, 'EMAIL_OPTIN_MINIMUM_AGE', 13),
```

Comment:
I left the 13 because it was there in the old code, and if it defaults to None then parental controls get switched off. It does feel a little odd to have the 13 left sprinkled through the code, so I'd be happy to remove it if that seems reasonable.